### PR TITLE
Use uniform prefix for all of the resources

### PR DESCRIPTION
**Pull request by yasun1**
[OCM-8258](https://issues.redhat.com//browse/OCM-8258): 
We saw that even though I started two pre submit jobs with 20m intervals, I see the two jobs use the same namespace ci-op-d3pnt1s1. We have resources that use the namespace as the prefix. The same namespace makes the resources mixed to be dirty to fail jobs.
Jobs:
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_rosa/2033/pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp/1793473488065925120
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_rosa/2033/pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-sts-byo-kms-oidc-fips/1793478999440101376

https://redhat-internal.slack.com/archives/CBN38N3MW/p1716434067529659

### CODE REVIEW

**Comment by yasun1** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-sts-byo-kms-oidc-fips pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-byo-kms-oidc-auditlog pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-non-sts periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-rosa-hcp-capi-private-stage-critical-f7

**Comment by yasun1** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-non-sts periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-rosa-hcp-capi-stage-critical-f2 pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-byo-kms-oidc-auditlog

File: ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh [hunk 1]
```diff
@@ -32,6 +32,7 @@ CLUSTER_SECTOR=${CLUSTER_SECTOR:-}
 ADDITIONAL_SECURITY_GROUP=${ADDITIONAL_SECURITY_GROUP:-false}
 NO_CNI=${NO_CNI:-false}
 CONFIGURE_CLUSTER_AUTOSCALER=${CONFIGURE_CLUSTER_AUTOSCALER:-false}
+CLUSTET_PREFIX=$(head -n 1 "${SHARED_DIR}/cluster-prefix")
```
Comment (radtriste) [inline ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:35]: `CLUSTER_PREFIX` ? :)

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-rosa-hcp-capi-stage-critical-f2

File: ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh (hunk 1 above)
Comment (yasun1) [reply ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:35]: @radtriste This variable is used by `cluster_name`, `account-roles`, and `operator-roles` in this script. I do not think of the right name for these usages, but I inherit from the step directly. Any suggestion?

Comment (yasun1) [reply ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh]: Updated.

**Comment by yasun1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-rosa-hcp-capi-stage-critical-f2

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**